criprobe/cri.py

```diff
@@ -1,14 +1,54 @@
 import re
+import serial
+import serial.tools.list_ports
 
 
 class CriProbe:
 
     def __init__(self, simulated=False):
-        # Perform autodetect here
-        # When no probes are detected we'll create CR-100 and CR-250 simulated probes
-        # these simulated probes will primarily be used for unit testing
+
+        # Autodetects CRI probe/s.
+
         if simulated:
             self.probes = [{'Type': 'CR-100'},
                            {'Type': 'CR-250'}]
         else:
-            self.probes = None
+            self.probes = []
+            ports = serial.tools.list_ports.comports()
+            for port in ports:
+                if re.search(r'A\d{6}', port.device):
+                    with serial.Serial(port.device, 115200, timeout=1) as cri_probe:
+                        probe_info = {}
+
+                        cri_probe.write(b'RC ID\r\n')
+                        probe_result = cri_probe.readline()
+                        id = re.search(r'(A\d{5})', str(probe_result))
+                        if id:
+                            probe_info['ID'] = str(id.group(1))
+                        else:
+                            raise RuntimeError('CRI Probe ID Not Found')
+
+                        cri_probe.write(b'RC Model\r\n')
+                        probe_result = cri_probe.readline()
+                        model = re.search(r'(CR-\d{3})', str(probe_result))
+                        if model:
+                            probe_info['Model'] = str(model.group(1))
+                        else:
+                            raise RuntimeError('CRI Probe Model Not Found')
+
+                        cri_probe.write(b'RC InstrumentType\r\n')
+                        probe_result = cri_probe.readline()
+                        instrument_type = re.search(r'(\d)', str(probe_result))
+                        if instrument_type:
+                            reg_type = instrument_type.group(1)
+                            if int(reg_type) == 0:
+                                probe_type = 'Photometer'
+                            elif int(reg_type) == 1:
+                                probe_type = 'Colorimeter'
+                            elif int(reg_type) == 2:
+                                probe_type = 'Spectroradiometer'
+                        else:
+                            raise RuntimeError('CRI Probe Type Not Found')
+                        probe_info['Type'] = probe_type
```

Comment:
This line is not in the correct place - needs to be moved up above the else and indented.